### ISSUE TITLE
Does podio supports c++17 standard?

### ISSUE THREAD

**Issue by peremato** [AIDASoft/podio#62]
When building podio on MacOS configured with `-DCMAKE_CXX_STANDARD=17` I get the following compilation errors:
```
[ 98%] Built target write_ascii
In file included from /Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/unittest.cpp:7:
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:3453:14: error: no template named 'auto_ptr' in namespace 'std'
        std::auto_ptr<StreamBufBase> m_streamBuf
        ~~~~~^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:3601:14: error: no template named 'auto_ptr' in namespace 'std'
        std::auto_ptr<IStream const> m_stream
        ~~~~~^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:3592:28: error: cannot initialize return object of type 'const Catch::IStream *' with an rvalue of type 'Catch::DebugOutStream *'
                    return new DebugOutStream()
                           ^~~~~~~~~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:5307:63: error: no matching function for call to 'getAllTestCasesSorted'
        std::vector<TestCase> matchedTestCases = filterTests( getAllTestCasesSorted( config ), testSpec, config )
                                                              ^~~~~~~~~~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:628:34: note: candidate function not viable: no known conversion from 'const Catch::Config' to 'const Catch::IConfig' for 1st argument
    std::vector<TestCase> const& getAllTestCasesSorted( IConfig const& config )
                                 ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:5335:63: error: no matching function for call to 'getAllTestCasesSorted'
        std::vector<TestCase> matchedTestCases = filterTests( getAllTestCasesSorted( config ), testSpec, config )
                                                              ^~~~~~~~~~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:628:34: note: candidate function not viable: no known conversion from 'const Catch::Config' to 'const Catch::IConfig' for 1st argument
    std::vector<TestCase> const& getAllTestCasesSorted( IConfig const& config )
                                 ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:5375:63: error: no matching function for call to 'getAllTestCasesSorted'
        std::vector<TestCase> matchedTestCases = filterTests( getAllTestCasesSorted( config ), testSpec, config )
                                                              ^~~~~~~~~~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:628:34: note: candidate function not viable: no known conversion from 'const Catch::Config' to 'const Catch::IConfig' for 1st argument
    std::vector<TestCase> const& getAllTestCasesSorted( IConfig const& config )
                                 ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6201:105: error: reference to type 'const Ptr<const Catch::IConfig>' could not bind to an rvalue of type 'Catch::Config *'
        Ptr<IStreamingReporter> reporter = getRegistryHub().getReporterRegistry().create( reporterName, config.get() )
                                                                                                        ^~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:5278:96: note: passing argument to parameter 'config' here
        virtual IStreamingReporter* create( std::string const& name, Ptr<IConfig const> const& config ) const = 0
                                                                                               ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6233:28: error: no viable conversion from 'Catch::Config *' to 'Ptr<const Catch::IConfig>'
        Ptr<IConfig const> iconfig = config.get()
                           ^         ~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:494:9: note: candidate constructor not viable: no known conversion from 'Catch::Config *' to 'const Catch::IConfig *' for 1st argument
        Ptr( T* p ) : m_p( p ){
        ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:498:9: note: candidate constructor not viable: no known conversion from 'Catch::Config *' to 'const Catch::Ptr<const Catch::IConfig> &' for 1st argument
        Ptr( Ptr const& other ) : m_p( other.m_p ){
        ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6355:17: error: no matching function for call to 'seedRng'
                seedRng( *m_config )
                ^~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:405:10: note: candidate function not viable: no known conversion from 'Catch::Config' to 'const Catch::IConfig' for 1st argument
    void seedRng( IConfig const& config )
         ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6358:21: error: no matching function for call to 'applyFilenamesAsTags'
                    applyFilenamesAsTags( *m_config )
                    ^~~~~~~~~~~~~~~~~~~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6262:10: note: candidate function not viable: no known conversion from 'Catch::Config' to 'const Catch::IConfig' for 1st argument
    void applyFilenamesAsTags( IConfig const& config ) {
         ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6431:26: error: no member named 'random_shuffle' in namespace 'std'
                    std::random_shuffle( sorted.begin(), sorted.end(), rng )
                    ~~~~~^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:504:17: error: cannot initialize object parameter of type 'const Catch::SharedImpl<Catch::IConfig>' with an expression of type 'Catch::Config'
                m_p->release()
                ^~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6289:9: note: in instantiation of member function 'Catch::Ptr<Catch::Config>::~Ptr' requested here
        Session()
        ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:508:17: error: cannot initialize object parameter of type 'const Catch::SharedImpl<Catch::IConfig>' with an expression of type 'Catch::Config'
                m_p->release()
                ^~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6315:26: note: in instantiation of member function 'Catch::Ptr<Catch::Config>::reset' requested here
                m_config.reset()
                         ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:496:17: error: cannot initialize object parameter of type 'const Catch::SharedImpl<Catch::IConfig>' with an expression of type 'Catch::Config'
                m_p->addRef()
                ^~~
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:512:17: note: in instantiation of member function 'Catch::Ptr<Catch::Config>::Ptr' requested here
            Ptr temp( p )
                ^
/Users/mato/Development/LCGSoft/build/podio-v00-09-02/src/podio/v00-09-02/tests/catch.hpp:6383:26: note: in instantiation of member function 'Catch::Ptr<Catch::Config>::operator=' requested here
                m_config = new Config( m_configData )
                         ^
14 errors generated.
make[6]: *** [tests/CMakeFiles/unittest.dir/unittest.cpp.o] Error 1
make[5]: *** [tests/CMakeFiles/unittest.dir/all] Error 2
make[4]: *** [all] Error 2

```

**Comment by peremato** [comment]
See https://github.com/AIDASoft/podio/pull/63

**Comment by andresailer** [comment]
Fixed by #63 ?
